lib/diagnostic.rb

```diff
@@ -13,28 +13,31 @@ def longest(enum)
   # enumerable supplied (i.e. apply the `+` operator)
   # Assume each entry implements a `+` method
   def sum(enum)
-    # Your code goes here
+    enum.each do |prev_item, nex_item|
+      prev_item += nex_item
+    end
```

Comment:
You're spinning your wheels with this. `prev_item` is not defined outside of the block, so data isn't being stored from one iteration to the next, and `.each` returns `undefined`, so the method will return `undefined` in any case.